src/do-you-want-to-stop.py

```diff
@@ -1,3 +1,10 @@
 
 # You should write a loop around this question, and keep asking until
 # the answer is "yes" (lower case).
+i="no"
+while i == "no":
+    i = input("Do you want to stop? ")
+    if i == 'yes':
+        break
+    else:
+        i="no"
```

Comment:
Generally, I would not use `i` as a variable that holds a string. By convention, `i` and `j` refer to integers, and experienced programmers will get confused if they refer to strings. That being said, it is an excellent solution.

You don't need the test for the `while` condition, though. You use `break` to leave the loop if you get `'yes'`, so you are really testing the same thing twice. You could instead do this:

```python
while True:
    x = input("Do you want to stop? ")
   if x == "yes":
       break
```

or you could do something like this:

```python
x = 'something'
while x != 'yes':
    x = input("Do you want to stop? ")
```